<launch>
  <arg name="rosbag_file" default="translation.bag" />
  <arg name="filename" default="$(find hand_motion_estimator)/sample/data/$(arg rosbag_file)" />
  <arg name="ROBOT" default="AERO" />
  <arg name="options" value="--clock --loop" />
  <arg name="manager" value="camera_remote_nodelet_manager" />

  <arg name="respawn" value="false" />
  <arg name="bond" value="" if="$(arg respawn)"/>
  <arg name="bond" value="--no-bond" unless="$(arg respawn)"/>
  <arg name="camera_name" default="openni_camera" />

  <param name="/use_sim_time" value="false" />

  <!-- play rosbag -->
  <node name="rosbag_play"
        pkg="rosbag"
        type="play"
        args="$(arg filename) $(arg options)" />

  <!-- view robot model -->
  <param name="robot_description"
         textfile="$(find aero_description)/robots/aero.urdf" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)"
        args="manager"  output="screen"/>

  <!-- expand compressed rgb image -->
  <node name="rgb_image_republish"
        pkg="image_transport" type="republish"
        output="screen" args="compressed raw">
    <remap from="in" to="/$(arg camera_name)/rgb/image_raw" />
    <remap from="out" to="/$(arg camera_name)/rgb/image_rect_color" />
  </node>

  <!-- expand compressed depth image -->
  <node name="depth_image_republish"
        pkg="image_transport" type="republish"
        output="screen" args="compressedDepth raw">
    <remap from="in" to="/$(arg camera_name)/depth_registered/image_raw" />
    <remap from="out" to="/$(arg camera_name)/depth_registered/image_rect" />
  </node>

  <!-- generate point cloud from rgb and depth image -->
  <node pkg="nodelet" type="nodelet" name="points_xyzrgb_hw_registered"
        args="load depth_image_proc/point_cloud_xyzrgb $(arg manager) $(arg bond)"
        respawn="$(arg respawn)">
    <remap from="rgb/image_rect_color"        to="/$(arg camera_name)/rgb/image_rect_color"/>
    <remap from="rgb/camera_info"             to="/$(arg camera_name)/rgb/camera_info"/>
    <remap from="depth_registered/image_rect" to="/$(arg camera_name)/depth_registered/image_rect"/>
    <remap from="depth_registered/points"     to="/$(arg camera_name)/point_cloud/cloud_registered/remote"/>
    <rosparam>
      queue_size: 100
    </rosparam>
  </node>

</launch>
